src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1323,50 +1329,53 @@ protected override void InvokeDefaultAction(string path)
 
             string resource = StringUtil.Format(FileSystemProviderStrings.InvokeItemResourceFileTemplate, path);
 
+            void StartProcessUsingHandler(string filename)
+            {
+                System.Diagnostics.Process process = new System.Diagnostics.Process();
+#if UNIX
+                // The file is possibly not an executable, so we try invoking the default program that handles this file.
+                const string quoteFormat = "\"{0}\"";
+                process.StartInfo.FileName = Platform.IsLinux ? "xdg-open" : /* OS X */ "open";
+                if (NativeCommandParameterBinder.NeedQuotes(filename))
+                {
+                    path = string.Format(CultureInfo.InvariantCulture, quoteFormat, filename);
```

Comment:
I thought we could remove the 'if' and each name quoted - "filename.txt", "file name.txt".